### DISCUSSION TITLE
How do we show filters (attributes) for search results?

### DISCUSSION THREAD

**Discussion by SpaiceVip** [Smile-SA/elasticsuite#3471] [Q&A]
I can't work out how to configure ElasticSuite (Magento 2.4.7) to show the filterable attributes for search results. 

It means our results can show hundreds, sometimes thousands, of results for a term but there is no way for the user to 'drill down' to the specifications or features they are interested in (e.g. colour, screen size, etc)

Can you help?

I've been advised that we need a separate, third-party, search extension but that doesn't make sense to us as it seems we're just installing  a new search module to work around ElasticSuite and most that we have looked at clash with the Smile module.

**Comment by romainruaud** [comment]
Hi @SpaiceVip 

Just set your attributes to "Use in Search Results Layered Navigation" in the Storefront tab of the attribute edition screen in Magento Back Office.

Regards

**Comment by SpaiceVip** [comment]
Hi @romainruaud 

Thanks for this. I looked into this and discovered that the Layered Navigation setting is already as you suggested it should be - so drilling through the categories shows all the filters in the sidebar navigation.

However, the "Is Filterable in Search" is set to no in all but the manufacturer attribute. And, manufacturer is the only filter that shows in our search results (see spaice.net)

So, that seems to be the problem, however, when we alter any of these other attributes to activate this filter there is no change to the frontend - only manufacturer shows.

I don't know whether Stock in the Channel have done something odd with the settings so they can't be configured but do you have any ideas on this?

Thanks,

Vipul

![2024_12_18 - Smile-SA ElasticSuite - Search Filters Off](https://github.com/user-attachments/assets/4c91b2c7-3b77-4652-bd4e-385922f1b170)

**Comment by SpaiceVip** [comment]
@romainruaud 

Hi,

Can you help on this?

We're finding that the only way to change filtered values (Is filterable = 1) is through directly altering the database and this works, the filters appear in search results. **But**, the magento admin panel will not save any changes.

And a bigger problem is that most of the Smile 'attributes' don't even show in the admin panel. Does ElasticSuite use it's own version of the attributes?

If you look at this page, none of the filters below manufacturer appear in the Magento attributes list but they're obviously being used by ElasticSuite

Thanks 

![image](https://github.com/user-attachments/assets/9f44467e-1788-4218-8e0c-9a7ca9590bc4)

**Reply by Bashev** [reply]
Probably you have too much brands, but not change php setting "max_input_vars", change the default value to 10000 and try again.

https://experienceleague.adobe.com/en/docs/commerce-knowledge-base/kb/troubleshooting/miscellaneous/php-settings-errors#max-input-vars-error-due-to-large-forms
